packages/panna-sdk/src/core/util/api-service.ts

```diff
@@ -227,6 +235,79 @@ export class PannaApiService {
       throw error;
     }
   }
+
+  /**
+   * Retrieves the status of an onramp.money session
+   *
+   * This function polls the onramp.money API to get the current status of a fiat-to-crypto
+   * onramp session. The status includes transaction details, amounts, and any error messages.
+   *
+   * @param params - Parameters for retrieving the session status
+   * @param params.sessionId - The onramp.money session identifier
+   * @param params.authToken - Optional JWT token for authentication
+   * @returns Promise resolving to the session status with all transaction details
+   * @throws Error if the session ID is invalid or network request fails
+   */
+  public async getSessionStatus(
+    params: GetSessionStatusParams
+  ): Promise<SessionStatusResult> {
+    const { baseUrl, isMockMode } = this.config;
+    const { sessionId, authToken } = params;
+
+    if (!sessionId) {
+      throw new Error('Session ID is required');
+    }
+
+    if (isMockMode) {
+      await new Promise((resolve) => setTimeout(resolve, 500));
+
+      if (sessionId.includes('created')) {
+        return mockSessionStatusCreated;
+      } else if (sessionId.includes('pending')) {
+        return mockSessionStatusPending;
+      } else {
+        return mockSessionStatusCompleted;
+      }
+    }
```

Comment:
Ideally this section shouldn't exist.
If we still want to maintain this, please import and use the enum instead.
```suggestion
    if (isMockMode) {
      await new Promise((resolve) => setTimeout(resolve, 500));

      if (sessionId.includes(OnrampMoneySessionStatusEnum.Created)) {
        return mockSessionStatusCreated;
      } else if (sessionId.includes(OnrampMoneySessionStatusEnum.Pending)) {
        return mockSessionStatusPending;
      } else {
        return mockSessionStatusCompleted;
      }
    }
```